design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,474 @@
+metadata:
+  name: secrets-access-binding
+  namespace: app-namespace
+subjects:
+  - kind: ServiceAccount
+    name: conjur-sa-1
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: Role
+  name: secrets-access
+  apiGroup: rbac.authorization.k8s.io
+```
+
+#### Code changes
+
+Although we will now offer "K8S_SECRET_LABELS" for easy filtering on K8s Secrets, we will still support the "K8S_SECRET" environment variable if the customer prefers to list the K8s Secrets they need values for from Conjur. 
+
+### Enhancements
+
+As optional (yet important) enhancements, we suggest the following:
+
+1. #### Differentiate between deployment types using K8s API
+
+   **Current solution**: Customer supplies this value in `CONTAINER_MODE` env var to Secrets Provider.
+
+   **Enhanced solution**: Use K8s API `GET namespaces/{namespace}/pods/{pod}` to get the pod's manifest and derive the deployment used for Secrets Provider.
+
+   **Motivation**: better UX, no need to maintain another env var. Also, ensures correct value and prevent mistakes.
+   **Requirements**: Add `get` rights for `pods` to Secrets Provider's service account in K8s.
+
+2. #### Get `MY_POD_NAME` `MY_POD_NAMESPACE`
+
+   **Current solution**: Customer supplies these values using [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/) for `MY_POD_NAME` and `MY_POD_NAMESPACE` env vars in Secrets Provider manifest.
+
+   **Enhanced solution**: Get pod's namespace from `/var/run/secrets/kubernetes.io/serviceaccount/namespace` file inside the container.
+   Get pod's name from `HOSTNAME` env var (as documented in [K8s docs](https://kubernetes.io/docs/concepts/containers/container-environment/#container-information)) 
+
+   **Motivation**: better UX, no need to maintain another env vars.
+
+   **Limitation**: `HOSTNAME` is the pod's name only if customer didn't [supply a hostname explicitly](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields).
+
+   **Solution**: Use `HOSTNAME` if `MY_POD_NAME` is not supplied.
+
+3. #### Do not update K8s secret with the same value
+
+   **Current solution**: For each K8s secret, pull its values from Conjur and write them into the K8s secret.
+
+   **Enhanced solution**: Before writing the K8s secret, compare it with the existing value. If the value is the same, skip its writing.
+
+   **Motivation**:
+
+   1. Less stress on K8s API in case of many k8s secrets that are already updated.
+   2. Remove redundant notifications because every change to K8s resource sends a notification to all watchers.
+   3. Improved performance due to less API calls
+
+### Order of Deployment
+
+To ensure that on first run our Secrets Provider runs first, we will request from the customer follow the following setup order:
+
+1. Add all necessary K8s Secrets and their labels to the Secrets Provider manifest
+
+2. Run the Secrets Provider Job
```

Comment:
Please, provide a sample manifest of this job - including it time defaults (once in how long it runs)

Reply:
Detailed in doc below (: